code/__HELPERS/unsorted.dm

```diff
@@ -1550,13 +1561,18 @@ GLOBAL_DATUM_INIT(dview_mob, /mob/dview, new)
 		sleep(0.6)
 
 	if(orbiting == A) //make sure we haven't started orbiting something else.
-		orbiting = null
+
 		SpinAnimation(0, 0)
-		transform = cached_transform
+		stop_orbit()
 
 
 
 /atom/movable/proc/stop_orbit()
+	if (orbiting.orbiters)
+		orbiting.orbiters -= (src)
+		if (orbiting.orbiters.len == 0)
+			qdel(orbiting.orbiters)
+			orbiting.orbiters = null
```

Comment:
```suggestion
			orbiting.orbiters = null
```

Just null it out. Calling `qdel()` on a list automatically pipes it to a hard delete.